src/nvim/message.c

```diff
@@ -3240,46 +3230,41 @@ int vim_vsnprintf(char *str, size_t str_m, char *fmt, va_list ap, typval_T *tvs)
         }
       }
 
-      /* parse 'h', 'l' and 'll' length modifiers */
-      if (*p == 'h' || *p == 'l') {
+      // parse 'h', 'l', 'll' and 'z' length modifiers
+      if (*p == 'h' || *p == 'l' || *p == 'z') {
         length_modifier = *p;
         p++;
-        if (length_modifier == 'l' && *p == 'l') { /* double l = long long */
-          length_modifier = '2';                   /* double l encoded as '2' */
+        if (length_modifier == 'l' && *p == 'l') { // ll, encoded as 2
+          length_modifier = '2';
           p++;
         }
       }
+
       fmt_spec = *p;
 
-      /* common synonyms: */
+      // common synonyms
       switch (fmt_spec) {
-      case 'i': fmt_spec = 'd'; break;
-      case 'D': fmt_spec = 'd'; length_modifier = 'l'; break;
-      case 'U': fmt_spec = 'u'; length_modifier = 'l'; break;
-      case 'O': fmt_spec = 'o'; length_modifier = 'l'; break;
-      case 'F': fmt_spec = 'f'; break;
-      default: break;
+        case 'i': fmt_spec = 'd'; break;
+        case 'D': fmt_spec = 'd'; length_modifier = 'l'; break;
+        case 'U': fmt_spec = 'u'; length_modifier = 'l'; break;
+        case 'O': fmt_spec = 'o'; length_modifier = 'l'; break;
+        case 'F': fmt_spec = 'f'; break;
+        default: break;
       }
 
-      /* get parameter value, do initial processing */
+      // get parameter value, do initial processing
       switch (fmt_spec) {
-      /* '%' and 'c' behave similar to 's' regarding flags and field widths */
-      case '%':
-      case 'c':
-      case 's':
-      case 'S':
+      // '%' and 'c' behave similar to 's' regarding flags and field widths
+      case '%': case 'c': case 's': case 'S':
```

Comment:
Yes, that's non-standard. Recommended way is one-line-per-case.
I just did it to match other places in the same function where that is done.
Though I cleaned up a bit, there are still a lot of style violations in this code (specially in switch statements, which don't indent cases, for example). But they do because there are lots of them and with a lot of nesting. So, it can make sense in this function.
If a future PR completely reformats this to guide-blessed style, it can take care of that.
I could do it here if it bothers too much, though.